### ISSUE TITLE
Add a speedometer widget

### ISSUE THREAD

**Issue by ianthetechie** [stadiamaps/ferrostar#247]
This should display the user's current speed (which we should be getting from GPS), when it is available. 

We had a prospective user say this was a hard requirement for them, and they are migrating from some archaic Mapbox SDKs. I have not personally used this, so I don't have a good intuition for what it should look like and how it interacts with the speed _limit_ view that we already have.

I'll push for some better definition, but in the meantime, perhaps @Archdoog or @haysmike have some opinions / screenshots / mockups to share (be sure to note source and license if it's not a permissive one so we don't accidentally run into issues).

**Comment by Archdoog** [comment]
If we wanted a built in method, we should probably attach it to the speed limit signs like several navigation apps do (e.g. Google Maps, [Mapbox Navigation](https://cdn.prod.website-files.com/6050a76fa6a633d5d54ae714/6596080d45084a14ff2582b2_img-hero.webp), etc). It could be customizable in the default view where you you can configure:

```swift
enum SpeedView {
    case speedAndLimit
    case limit
    case speed
}
```

As an aside, I would also say this is also extremely easy to build on their own once we surface a formatted speed value. A significant number of navigation apps have their own customized widget for this. (e.g. [Waze](https://mobbin.com/screens/69d20152-a18c-4304-9ba1-b307183b3450?workspace=ada1cdf9-e8ef-4a1b-bd10-69b09b592f83&utm_source=share_link&utm_medium=share&utm_campaign=screen_sharing)). I've already got a bespoke speedometer in my app that's part of a custom arrival view. Last note here, it should be very easy to place this on the navigation views using the empty grid spots.

**Comment by ianthetechie** [comment]
Yeah, I've been looking around at other designs and rather like the option to have a speedometer "attached" with the speed limit sign if desired. I'll whip up something for this next week since it's probably pretty easy.

**Comment by ianthetechie** [comment]
See https://github.com/stadiamaps/ferrostar/pull/360#discussion_r1847505234 for Android ideas. We'll eventually incorporate this probably.

Note from @Archdoog: there may be an issue with GPS updates never delivering a "zero" speed update when you stop. Needs further investigation. Currently observed on Android; unsure of iOS.
